### ISSUE TITLE
Question about SC2173

### ISSUE THREAD

**Issue by QuantumLibet** [koalaman/shellcheck#2424]
#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

code snippet.sh
```sh
#!/usr/local/env bash
trap 'echo "trapped"; exit 1' KILL
while true; do { echo "working"; sleep 1; } done
```

output after starting the snippet.sh, and killing it after a few seconds (from another terminal):
```sh
$> ./snippet.sh 
working
working
working
working
trapped
```

#### Here's what shellcheck currently says:

https://github.com/koalaman/shellcheck/wiki/SC2173
In particular: "…SIGKILL and SIGSTOP can not be caught…"

#### Here's what I wanted or expected to see:

According to the rationale of SC2173, I expected the snipped to be killed without the kill being trap'ed. If so, the last line of output should be 'working'. Instead, it is 'trapped' which means the trap is in fact recognized and the trap routine runs.

Is this a new behaviour of bash version 5.1.12(1)-release (x86_64-apple-darwin19.6.0), or is this the result of the wording "Trying to trap this signal has undefined results."?

Thank you.

**Comment by QuantumLibet** [comment]
There is a circle at the top, which says "1 of 2 tasks". Haven't seen this before.

I assume this is related to the two checkmarks in the post. I also assume that this means I'd have to check both boxes for the circle to fill.

Since this topic is open for almost two years by now, I assume this is because it is either to nobody interesting, than to me, or it is because it waits for the second checkbox to be checked. 

Since I do not know what to do to check the box that says "(i.e. the latest commit)" and since I do not want to claim I've done something that I clearly have no idea how to, I don't see how this can go forward.

So I'll close this point. No sense leaving it open.

**Comment by brother** [comment]
> or it is because it waits for the second checkbox to be checked.

That's for sure not the reason at least. 
The circle is indeed connected to the checkboxes but humans are evaluating all the things around here and can draw conclusions from the missing check mark if needed be.

Two years is a long time indeed, but it's far from the longest standing issues in this project. It's time consuming to do hobby projects at the side of other duties.

**Comment by brother** [comment]
Tried the example just for fun. I don't have a /usr/local/env so replaced the shebang with /bin/bash.

```shell
brother ~$ bash --version
GNU bash, version 5.2.15(1)-release (x86_64-pc-linux-gnu)
Copyright (C) 2022 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

brother ~$ cat /tmp/test.sh
#!/bin/bash

trap 'echo "trapped"; exit 1' KILL

echo "Running shell in pid $$"

while true; do { echo "working"; sleep 1; } done
brother ~$ /tmp/test.sh
Running shell in pid 86813
working
working
working
working
working
working
working
Killed
brother ~$ /tmp/test.sh
Running shell in pid 86967
working
working
working
working
working
Terminated
```
(second attempt was a standard TERM just to illustrate)


I don't know if mac handles signals different than linux, it's possible. But according to the wiki it should not be possible to trap KILL and STOP.

[POSIX states](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#trap):
> Setting a trap for SIGKILL or SIGSTOP produces undefined results.
